src/main/java/it/gov/pagopa/pu/fileshare/exception/FileshareExceptionHandler.java

```diff
@@ -0,0 +1,50 @@
+package it.gov.pagopa.pu.fileshare.exception;
+
+import it.gov.pagopa.pu.fileshare.dto.generated.FileshareErrorDTO;
+import it.gov.pagopa.pu.fileshare.dto.generated.FileshareErrorDTO.CodeEnum;
+import it.gov.pagopa.pu.fileshare.exception.custom.InvalidFileException;
+import jakarta.servlet.http.HttpServletRequest;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.core.Ordered;
+import org.springframework.core.annotation.Order;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+
+/**
+ * A class exception that handles errors related to workflows.
+ *
+ */
+@RestControllerAdvice
+@Slf4j
+@Order(Ordered.HIGHEST_PRECEDENCE)
+public class FileshareExceptionHandler {
+
+  @ExceptionHandler({InvalidFileException.class})
+  public ResponseEntity<FileshareErrorDTO> handleInvalidFileError(RuntimeException ex, HttpServletRequest request){
+    return handleFileshareErrorException(ex, request, HttpStatus.INTERNAL_SERVER_ERROR, CodeEnum.INVALID_FILE);
```

Comment:
why a 500 status code?
this exception is intended for an invalid file, it should return 400